Simulator/Recorders/XmlRecorder.h

```diff
@@ -57,16 +55,61 @@ class XmlRecorder : public IRecorder {
    ///  Registered to OperationManager as Operation::printParameters
    virtual void printParameters() override;
 
+   /// Store the neuron number and all the events of this single neuron
+   void registerVariable(string varName, EventBuffer &recordVar) override;
+
+   ///@{
+   /** These methods are intended only for unit tests */
+   // constructor only for unit test
+   XmlRecorder(std::string fileName_)
+   {
+      resultFileName_ = fileName_;
+   }
+
+   // Getter method for neuronName (only included during unit tests)
+   std::string getNeuronName() const
+   {
+      return neuronName;
+   }
+   // Getter method for singleNeuronEvents_ (only included during unit tests)
+   EventBuffer &getSingleNeuronEvents() const
+   {
+      return *singleNeuronEvents_;
+   }
+   // Getter method for single_neuron_History_ (only included during unit tests)
+   std::vector<uint64_t> getHistory() const
+   {
+      return single_neuron_History_;
+   }
+   ///@}
+
 protected:
+   // variable neuronName records the number of a single neuron
+   string neuronName;
+
+   // The address of the registered variable
+   // As the simulator runs, the values will be updated
+   // It can records all events of a single neuron in each epoch
+   shared_ptr<EventBuffer> singleNeuronEvents_;
+
+   // history of accumulated event for a single neuron
+   std::vector<uint64_t> single_neuron_History_;
```

Comment:
Why use the `_` naming style for this one member variable?

Reply:
I should name it with singleNeuronHistory_. Sorry for this mistake